Exploratory Data Analysis

In [1]:
import pandas as pd

In [2]:
Movies2000 = pd.read_csv('data/final_tmdb_data_2000.csv.gz')
Movies2001 = pd.read_csv('data/final_tmdb_data_2001.csv.gz')

In [3]:
Movies = pd.concat([Movies2000,Movies2001])

In [4]:
Movies = Movies[((Movies.budget > 0) | (Movies.revenue > 0))]
Movies.info()

<class 'pandas.core.frame.DataFrame'>
Index: 664 entries, 1 to 1575
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                664 non-null    object 
 1   adult                  664 non-null    float64
 2   backdrop_path          558 non-null    object 
 3   belongs_to_collection  111 non-null    object 
 4   budget                 664 non-null    float64
 5   genres                 664 non-null    object 
 6   homepage               78 non-null     object 
 7   id                     664 non-null    float64
 8   original_language      664 non-null    object 
 9   original_title         664 non-null    object 
 10  overview               661 non-null    object 
 11  popularity             664 non-null    float64
 12  poster_path            641 non-null    object 
 13  production_companies   664 non-null    object 
 14  production_countries   664 non-null    object 
 15  release_da

1. How many movies had at least some valid financial information (values > 0 for budget OR revenue)?

~664

In [7]:
# 2. How many movies are there in each of the certification categories (G/PG/PG-13/R)?
Movies.value_counts('certification')

certification
R        230
PG-13    134
PG        38
NR        21
G         15
Name: count, dtype: int64

In [8]:
# What is the average revenue per certification category?
Movies.groupby('certification')['revenue'].mean()

certification
G        1.173648e+08
NR       1.683156e+07
PG       1.021913e+08
PG-13    9.780680e+07
R        3.015843e+07
Name: revenue, dtype: float64

In [9]:
# What is the average budget per certification category?
Movies.groupby('certification')['budget'].mean()

certification
G        3.813333e+07
NR       9.184624e+06
PG       4.197361e+07
PG-13    4.303103e+07
R        1.831437e+07
Name: budget, dtype: float64